chore: disable sleep mode when detecting video format

ym opened the pull request.

ym committed:
chore: disable sleep mode when detecting video format

ym committed:
fix: streaming mutex

ym committed:
fix: return value not checked

ym committed:
feat: add video quality factor config

shakdoesgithub commented on internal/native/cgo/video.c:
```diff
@@ -51,6 +53,34 @@ RK_U64 get_us()
+        return;
+    }
+    sleep_mode_available = true;
+    ensure_sleep_mode_disabled();
```
i feel like this function should just do the detection of the sleep mode and the ensuring of the sleep mode being disabled should be called by the consumer of this function. I feel like this function is doing too much.

adamshiervani committed:
feat: add power saving mode toggle in hardware settings

Implemented a new feature to enable or disable HDMI sleep mode, allowing users to reduce power consumption when the device is not in use. Added state management for the power saving setting and integrated it with the existing settings page.

ym committed:
chore: add logging when disabling HDMI sleep mode

thinkafterbefore committed:
Fix VIDIOC_S_FMT EBUSY error on video stream restart

V4L2 buffers were not explicitly freed before closing the device, causing VIDIOC_S_FMT to fail with EBUSY on restart. Added  VIDIOC_REQBUFS(count=0) call in cleanup to free buffer queue synchronously before close().

This ensures subsequent opens can set format immediately without EBUSY errors, eliminating 100ms+ retry delays.

ym committed:
fix: why set EDID forever

ym committed:
chore: do not restart video streaming if the format is the same

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/jetkvm/kvm?pullRequest=887) <br/>All committers have signed the CLA.

IDisposable commented on internal/native/video.go:
```diff
@@ -69,12 +73,32 @@ func (n *Native) VideoSleepModeSupported() bool {
+// it's used to ensure that only one change is made to the video stream at a time.
+// as the change usually requires to restart video streaming
+// TODO: check video streaming status instead of using a hardcoded timeout
+func (n *Native) useExtraLock(fn func() error) error {
```
Could this queuing-up be better handled by pushing the change into a channel to be handled inside a go-routine that serves up the changes in-order?

thinkafterbefore committed:
fix: update power saving mode condition to include zero duration

IDisposable commented on internal/native/cgo/video.c:
```diff
@@ -301,11 +349,29 @@ static void *venc_read_stream(void *arg)
 
 uint32_t detected_width, detected_height;
-bool detected_signal = false, streaming_flag = false;
+bool detected_signal = false, streaming_flag = false, streaming_stopped = true;
```
Shouldn't we declare `streaming_stopped` as `volatile` so the compiler knows that another thread might be mucking with them? (`streaming_flag` is protected by mutex, so should be fine, but wouldn't hurt to declare it `volatile` as well?)

IDisposable reviewed (commented):
I know this already merged, but I had a couple questions/comments
